js/modules/k6/grpc/client.go

```diff
@@ -108,6 +111,84 @@ func (c *Client) LoadProtoset(protosetPath string) ([]MethodInfo, error) {
 	return c.convertToMethodInfo(fdset)
 }
 
+func decryptPrivateKey(key, password []byte) ([]byte, error) {
+	block, _ := pem.Decode(key)
+	if block == nil {
+		return nil, fmt.Errorf("failed to decode PEM key")
+	}
+
+	blockType := block.Type
+	if blockType == "ENCRYPTED PRIVATE KEY" {
+		return nil, fmt.Errorf("encrypted pkcs8 formatted key is not supported")
+	}
+	/*
+	   Even though `DecryptPEMBlock` has been deprecated since 1.16.x it is still
+	   being used here because it is deprecated due to it not supporting *good* crypography
+	   ultimately though we want to support something so we will be using it for now.
+	*/
+	decryptedKey, err := x509.DecryptPEMBlock(block, password) //nolint:staticcheck
+	if err != nil {
+		return nil, err
+	}
+	key = pem.EncodeToMemory(&pem.Block{
+		Type:  blockType,
+		Bytes: decryptedKey,
+	})
+	return key, nil
+}
+
+func buildTLSConfig(certificate, key []byte, caCertificates [][]byte) (*tls.Config, error) {
+	var cp *x509.CertPool
+	if len(caCertificates) > 0 {
+		cp, _ = x509.SystemCertPool()
+		for i, caCert := range caCertificates {
+			if ok := cp.AppendCertsFromPEM(caCert); !ok {
+				return nil, fmt.Errorf("failed to append ca certificate [%d] from PEM", i)
+			}
+		}
+	}
+	cert, certerr := tls.X509KeyPair(certificate, key)
+	if certerr != nil {
+		return nil, fmt.Errorf("failed to append certificate from PEM")
+	}
+	return &tls.Config{
+		RootCAs:      cp,
+		Certificates: []tls.Certificate{cert},
+		MinVersion:   tls.VersionTLS12,
```

Comment:
None; whatsoever. Nice catch, I believe this, again, would be a copy/paste issue (but this time from something of my own internally)

Reply:
I've added a `//nolint:golint,gosec` comment for this particular line of code. I've also modified the tls.Config that is created so that it inherits `CipherSuites`, `InsecureSkipVerify`, `MinVersion`, `MaxVersion`, and `Renegotiation` from the `state.TLSConfig.Clone()`